bin/gen_peptides.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+
+import sys
+import argparse
+import pandas as pd
+
+from Bio.SeqIO.FastaIO import SimpleFastaParser
+from Fred2.Core import Allele, Peptide, Protein,generate_peptides_from_proteins
+
+
+parser = argparse.ArgumentParser("Generating peptides from protein sequences.")
+parser.add_argument('-i', '--input', metavar='FILE', type=argparse.FileType('r'), help = 'FASTA filename containing proteins.')
+parser.add_argument('-o', '--output', metavar='FILE', type=argparse.FileType('w'), help='Output file: peptides.')
+parser.add_argument('-min', '--min_length', metavar='N', type=int, help='Min. length of peptides that will be generated.')
+parser.add_argument('-max', '--max_length', metavar='N', type=int, help='Max. length of peptides that will be generated.')
+args = parser.parse_args()
+
+
+
+def _read_fasta(file, in_type=Protein):
+    # split at first whitespace and use short ID
+
+    collect = set()
+    # iterate over all FASTA entries:
+    for _id, seq in SimpleFastaParser(file):
+        # generate element:
+        _id = _id.split(" ")[0]
+
+        try:
+            collect.add(in_type(seq.strip().upper(), transcript_id=_id))
+        except TypeError:
+            collect.add(in_type(seq.strip().upper()))
+    return list(collect)
+
+
+proteins = _read_fasta(args.input, in_type=Protein)
+
+c = 0
+for k in range(args.min_length, args.max_length+1):
+    peptides = generate_peptides_from_proteins(proteins, k)
+    # get proteins and corresponding counts
+    pd_peptides = pd.DataFrame(
+        [ (str(pep), ','.join([ prot.transcript_id.split(' ')[0] for prot in pep.get_all_proteins() ]), ','.join([ str(len(pep.proteinPos[prot.transcript_id])) for prot in pep.get_all_proteins() ])) for pep in peptides ],
+        columns = ['sequence', 'proteins', 'counts']
```

Comment:
Should we call the column `protein_identifiers` or something like that?

Reply:
changed to `protein_ids`